standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/PersistenceManagerProvider.java

```diff
@@ -499,6 +514,14 @@ public static PersistenceManager getPersistenceManager(boolean forCompactor) {
     }
   }
 
+  public static DatabaseProduct getDatabaseProduct() {
+    if (databaseProduct == null) {
+      throw new RuntimeException(
```

Comment:
Same here, is it possible to use some specialized Metastore runtime exception?

Reply:
maybe change to `IllegalStateException`?

Reply:
sure